x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/validate.test.ts

```diff
@@ -140,5 +155,33 @@ describe('validate', () => {
       };
       expect(validatedOrError).toEqual(expected);
     });
+
+    test('it should do a validation correctly of a rule id with ruleStatus passed in', () => {
+      const ruleStatus = getFindResultStatus();
+      const ruleAlert = getResult();
+      const validatedOrError = transformValidateBulkError('rule-1', ruleAlert, null, ruleStatus);
+      expect(validatedOrError).toEqual({
+        ...ruleOutput(),
+        status: 'succeeded',
+        status_date: '2020-02-18T15:26:49.783Z',
+        last_success_at: '2020-02-18T15:26:49.783Z',
+        last_success_message: 'succeeded',
+      });
+    });
+
+    test('it should return error object if "alert" is not expected alert type', () => {
+      const ruleAlert = getResult();
+      // @ts-expect-error
+      delete ruleAlert.alertTypeId;
+      const validatedOrError = transformValidateBulkError('rule-1', ruleAlert);
+      const expected: BulkError = {
+        error: {
+          message: 'Internal error transforming',
+          status_code: 500,
+        },
+        rule_id: 'rule-1',
+      };
+      expect(validatedOrError).toEqual(expected);
```

Comment:
nit: Through out this test I would recommend just weakening your types to avoid the `ts-expect-error` (but I do appreciate you using `ts-expect-error` 👍  like so:

```ts
    test('it should return error object if "alert" is not expected alert type', () => {
      const { alertTypeId, ...ruleAlert } = getResult();
      const validatedOrError = transformValidateBulkError('rule-1', ruleAlert);
      const expected: BulkError = {
        error: {
          message: 'Internal error transforming',
          status_code: 500,
        },
        rule_id: 'rule-1',
      };
      expect(validatedOrError).toEqual(expected);
    });
```  

Either way works, I mostly use this way to avoid the newer TS errors on deleting things that used to not be there.